draft-barnes-acme.md

```diff
@@ -1212,66 +1212,74 @@ If all of the above verifications succeed, then the validation is successful.  I
 
 The Domain Validation with Server Name Indication (DVSNI) validation method aims to ensure that the ACME client has administrative access to the web server at the domain name being validated, and possession of the private key being authorized.  The ACME server verifies that the operator can reconfigure the web server by having the client create a new self-signed challenge certificate and respond to TLS connections from the ACME server with it.
 
-The challenge proceeds as follows: The ACME server sends the client a random value R and a nonce used to identify the transaction.  The client responds with another random value S.  The server initiates a TLS connection on port 443 to one or more of the IPv4 or IPv6 hosts with the domain name being validated.  In the handshake, the ACME server sets the Server Name Indication extension set to "\<nonce\>.acme.invalid".  The TLS server (i.e., the ACME client) should respond with a valid self-signed certificate containing both the domain name being validated and the domain name "\<Z\>.acme.invalid", where Z = SHA-256(R &#124;&#124; S).
+The challenge proceeds as follows: The ACME server sends the client a random value R, a nonce used to identify the transaction, and an array of acceptable ports for which the challenge may be performed.  The client responds with the desired port and key used within the test certificate.  The server initiates a TLS connection on the client's chosen port to one or more of the IPv4 or IPv6 hosts with the domain name being validated.  In the handshake, the ACME server sets the Server Name Indication extension set to "\<nonce\>.acme.invalid".  The TLS server (i.e., the ACME client) should respond with a valid self-signed certificate containing the domain name "\<R\>[0:32].\<R\>[32:64].acme.invalid".
 
-The ACME server's Challenge provides its random value R, and a random nonce used to identify the transaction:
+The ACME server's Challenge provides the following parameters:
 
 type (required, string):
 : The string "dvsni"
 
 r (required, string):
-: A random 32-byte octet, Base64-encoded
+: A random 32-byte octet, hex-encoded
 
 nonce (required, string):
 : A random 16-byte octet string, hex-encoded (so that it can be used as a DNS label)
```

Comment:
Assuming there isn't any further opposition, I will remove the SNI/cert complexity altogether.  After thinking about it for awhile longer, it will simplify the deployment in particular [web servers](http://doc.lighttpd.net/lighttpd2/mod_gnutls.html) and it does simplify the threat model.

Thanks